include/sonar_driver/OculusDriverNode.h

```diff
@@ -0,0 +1,66 @@
+
+#pragma once
+
+#include <stdint.h>
+#include <vector>
+#include <functional>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "std_msgs/msg/header.hpp"
+#include "sensor_msgs/msg/image.hpp"
+#include "sensor_msgs/msg/fluid_pressure.hpp"
+#include "sensor_msgs/msg/temperature.hpp"
+#include "geometry_msgs/msg/vector3_stamped.hpp"
+
+#include <sonar_driver_interfaces/msg/sonar_configuration.hpp>
+#include <sonar_driver_interfaces/msg/sonar_configuration_change.hpp>
+#include <sonar_driver_interfaces/msg/sonar_bearings.hpp>   
+
+
+#include <sonar_driver/sonardevices/Sonar.h>
+#include <sonar_driver/sonardevices/OculusSonar.h>
+
+class OculusDriverNode : public rclcpp::Node
+{
+public:
+    OculusDriverNode(const std::string& nodeName);
```

Comment:
@EulersBoiler Instead of instantiating a sonar inside the node directly, we should supply the dependency via the constructor. This allows for easy substitution (Liskov substitution principle) and would be in line with the dependency inversion principle.